napalm_brocade/brocade.py

```diff
@@ -0,0 +1,428 @@
+    ReplaceConfigException, SessionLockedException, CommandErrorException
+
+import os
+import re
+from shutil import copyfile
+
+
+# TBD(shh) Put this in config file (oslo_config)
+EXPORT_HOST = "10.24.88.6"
+EXPORT_USER = "shh"
+EXPORT_PASSWORD = "ss"
+
+class BrocadeDriver(NetworkDriver):
+    """Napalm Driver for Vendor Brocade."""
+
+    def __init__(self, hostname, username, password, timeout=60,
+                 optional_args=None):
+        """
+        CTOR for the device.
+        """
+
+        if optional_args is None:
+            optional_args = {}
+
+        self.device = None
+        self.hostname = hostname
+        self.username = username
+        self.password = password
+        self.timeout = timeout
+        self.port = optional_args.get('port', 22)
+
+
+    def open(self):
+        """Open a connection to the device."""
+        try:
+            self.device = ConnectHandler(device_type='brocade_vdx',
+                                         ip=self.hostname,
+                                         port=self.port,
+                                         username=self.username,
+                                         password=self.password,
+                                         timeout=self.timeout,
+                                         verbose=True)
+        except Exception:
+            raise ConnectionException("Cannot connect to switch: %s:%s" \
+                                          % (self.hostname, self.port))
+
+    def close(self):
+        """Close the connection to the device."""
+        self.device.disconnect()
+
+    def cli(self, commands=None):
+        """
+        Execute a list of commands and return the output in a dictionary format using the
+        command as the key.
+
+        """
+        cli_output = dict()
+
+        if type(commands) is not list:
+            raise TypeError('Please enter a valid list of commands!')
+
+        for command in commands:
+            output = self.device.send_command(command)
+            if 'Invalid input detected' in output:
+                raise ValueError(
+                    'Unable to execute command "{}"'.format(command))
+            cli_output.setdefault(command, {})
+            cli_output[command] = output
+
+        return cli_output
+
+    def send_command(self, cmd):
+        """Send the cmd to the switch for execution."""
+        output = self.device.send_command(cmd)
+        if 'Invalid input detected' in output:
+            raise ValueError('Unable to execute command "{}"'.format(cmd))
+        return output
+
+    def get_environment(self):
+
+        environment = {}
+        cpu_cmd = 'show proc cpu'
+
+        output = self.device.send_command(cpu_cmd)
+        output = output.strip()
+        environment.setdefault('cpu', {})
+        environment['cpu'][0] = {}
+        environment['cpu'][0]['%load'] = 0.0
+
+        lines = output.splitlines()
+
+        while True:
```

Comment:
I have a problem with ```break``` statements:

Can define something like:

```
def _search_cpu_load(lines):

    cpu_regex = r'^.*One minute: (\d+\.\d+); Five.*$'
    for line in lines:
        if 'Realtime Statistics' in line:
            match = re.search(cpu_regex, line)
            if match:
                return match.group(1)
```

And use it:

```
cpu_usage = _search_cpu_load(lines)
if cpu_usage:
    environment['cpu'][0]['%load'] = cpu_usage
```